Source generator doesn't handle nullable value types for Options

## Bug Description

The Nuru source generator doesn't generate type conversion code for nullable value types (`long?`, `int?`, etc.) on `[Option]` properties.

## Current Behavior

For an option defined as:
```csharp
[Option("exclude-invoice", "e", Description = "Invoice ID to exclude")]
public long? ExcludeInvoiceId { get; set; }
```

The generator produces:
```csharp
string? excludeinvoiceid = null;
// ... parsing logic reads value as string ...

// Then assigns directly without conversion:
ExcludeInvoiceId = excludeinvoiceid,  // ERROR: Cannot convert string to long?
```

This results in:
```
error CS0029: Cannot implicitly convert type 'string' to 'long?'
```

## Expected Behavior

The generator should produce TryParse code for nullable value types, similar to how it handles non-nullable types:

```csharp
string? __excludeinvoiceid_raw = null;
// ... parsing logic ...

long? excludeinvoiceid = null;
if (__excludeinvoiceid_raw != null)
{
    if (long.TryParse(__excludeinvoiceid_raw, NumberStyles.Integer, CultureInfo.InvariantCulture, out long parsed))
    {
        excludeinvoiceid = parsed;
    }
    else
    {
        // Error: invalid value
    }
}

// Then assign:
ExcludeInvoiceId = excludeinvoiceid,
```

## Comparison

| Property Type | Generated Code | Status |
|--------------|----------------|--------|
| `string?` | Direct assignment | ✅ Works |
| `long` | `long.TryParse(...)` | ✅ Works |
| `long?` | Direct string assignment | ❌ Bug |
| `int?` | (presumably same issue) | ❌ Bug |

## Reproduction

1. Create an endpoint with a nullable value type option
2. Build the project
3. Observe the CS0029 compilation error in generated code

--- a/kanban/done/397-fix-source-generator-nullable-value-type-handling-for-options-gh-146.md
+++ b/kanban/done/397-fix-source-generator-nullable-value-type-handling-for-options-gh-146.md
@@ -1,0 +1,35 @@
+# Fix source generator nullable value type handling for Options (GH #146)
+
+## Description
+
+The Nuru source generator doesn't generate type conversion code for nullable value types (`long?`, `int?`, etc.) on `[Option]` properties.
+
+**Current Behavior:** Generator produces direct string assignment for nullable types like `long?`, causing CS0029 compilation error:
+```csharp
+ExcludeInvoiceId = excludeinvoiceid,  // ERROR: Cannot convert string to long?
+```
+
+**Expected Behavior:** Generator should produce TryParse code for nullable value types, similar to non-nullable types:
+```csharp
+long? excludeinvoiceid = null;
+if (__excludeinvoiceid_raw != null)
+{
+    if (long.TryParse(__excludeinvoiceid_raw, NumberStyles.Integer, CultureInfo.InvariantCulture, out long parsed))
+    {
+        excludeinvoiceid = parsed;
+    }
+}
+```
+
+## Checklist
+
+- [ ] Identify where option type conversion is generated in the source generator
+- [ ] Add logic to detect nullable value types (Nullable<T>)
+- [ ] Generate appropriate TryParse code for nullable types
+- [ ] Add tests for nullable option types (long?, int?, bool?, etc.)
+
+## Notes
+
+- Reference: https://github.com/TimeWarpEngineering/timewarp-nuru/issues/146
+- Affected types: `long?`, `int?`, and other nullable value types on Options
+- Non-nullable value types already work correctly with TryParse

--- a/source/timewarp-nuru-analyzers/generators/emitters/route-matcher-emitter.cs
+++ b/source/timewarp-nuru-analyzers/generators/emitters/route-matcher-emitter.cs
@@ -1115,32 +1115,55 @@ internal static class RouteMatcherEmitter
     string optionDisplay = option.LongForm is not null ? $"--{option.LongForm}" : $"-{option.ShortForm}";
 
     // Use shared type conversion mapping with TryParse for graceful error handling
+    bool isNullableValueType = typeConstraint.EndsWith('?');
     (string ClrType, string TryParseCondition)? conversion = TypeConversionMap.GetBuiltInTryConversion(baseType, rawVarName, varName);
 
     if (conversion is var (clrType, tryParseCondition))
     {
-      // Determine the default value to use
-      string defaultValue = option.DefaultValueLiteral ?? "default";
-
-      if (option.ParameterIsOptional || option.DefaultValueLiteral is not null)
+      // Handle nullable value types (long?, int?, etc.) - declare as nullable and only parse if value provided
+      if (isNullableValueType)
       {
-        // Optional option value OR has default: declare with default, error only if value provided but invalid
-        sb.AppendLine($"      {clrType} {varName} = {defaultValue};");
-        sb.AppendLine($"      if ({rawVarName} is not null && !{tryParseCondition})");
+        string tempVarName = $"__{varName}_parsed_{routeIndex}";
+        sb.AppendLine($"      {clrType}? {varName} = null;");
+        sb.AppendLine($"      if ({rawVarName} is not null)");
         sb.AppendLine("      {");
-        sb.AppendLine($"        app.Terminal.WriteLine($\"Error: Invalid value '{{{rawVarName}}}' for option '{optionDisplay}'. Expected: {baseType}\");");
-        sb.AppendLine("        return 1;");
+        sb.AppendLine($"        if ({tryParseCondition.Replace($"out {varName}", $"out {clrType} {tempVarName}", StringComparison.Ordinal)})");
+        sb.AppendLine("        {");
+        sb.AppendLine($"          {varName} = {tempVarName};");
+        sb.AppendLine("        }");
+        sb.AppendLine("        else");
+        sb.AppendLine("        {");
+        sb.AppendLine($"          app.Terminal.WriteLine($\"Error: Invalid value '{{{rawVarName}}}' for option '{optionDisplay}'. Expected: {baseType}\");");
+        sb.AppendLine("          return 1;");
+        sb.AppendLine("        }");
         sb.AppendLine("      }");
       }
       else
       {
-        // Required option value (no default): TryParse with error message on failure
-        sb.AppendLine($"      {clrType} {varName} = default;");
-        sb.AppendLine($"      if ({rawVarName} is null || !{tryParseCondition})");
-        sb.AppendLine("      {");
-        sb.AppendLine($"        app.Terminal.WriteLine($\"Error: Invalid value '{{{rawVarName} ?? \"(missing)\"}}' for option '{optionDisplay}'. Expected: {baseType}\");");
-        sb.AppendLine("        return 1;");
-        sb.AppendLine("      }");
+        // Non-nullable value types - existing logic
+        // Determine the default value to use
+        string defaultValue = option.DefaultValueLiteral ?? "default";
+
+        if (option.ParameterIsOptional || option.DefaultValueLiteral is not null)
+        {
+          // Optional option value OR has default: declare with default, error only if value provided but invalid
+          sb.AppendLine($"      {clrType} {varName} = {defaultValue};");
+          sb.AppendLine($"      if ({rawVarName} is not null && !{tryParseCondition})");
+          sb.AppendLine("      {");
+          sb.AppendLine($"        app.Terminal.WriteLine($\"Error: Invalid value '{{{rawVarName}}}' for option '{optionDisplay}'. Expected: {baseType}\");");
+          sb.AppendLine("        return 1;");
+          sb.AppendLine("      }");
+        }
+        else
+        {
+          // Required option value (no default): TryParse with error message on failure
+          sb.AppendLine($"      {clrType} {varName} = default;");
+          sb.AppendLine($"      if ({rawVarName} is null || !{tryParseCondition})");
+          sb.AppendLine("      {");
+          sb.AppendLine($"        app.Terminal.WriteLine($\"Error: Invalid value '{{{rawVarName} ?? \"(missing)\"}}' for option '{optionDisplay}'. Expected: {baseType}\");");
+          sb.AppendLine("        return 1;");
+          sb.AppendLine("      }");
+        }
       }
     }
     else

--- a/tests/timewarp-nuru-tests/routing/routing-25-nullable-value-type-options.cs
+++ b/tests/timewarp-nuru-tests/routing/routing-25-nullable-value-type-options.cs
@@ -1,0 +1,244 @@
+#!/usr/bin/dotnet --
+
+#if !JARIBU_MULTI
+return await RunAllTests();
+#endif
+
+#pragma warning disable RCS1163 // Unused parameter - expected in negative test cases
+
+namespace TimeWarp.Nuru.Tests.Routing
+{
+
+/// <summary>
+/// Tests for nullable value type options (long?, int?, etc.)
+/// Reproduces GitHub issue #146: Source generator doesn't handle nullable value types for Options
+/// </summary>
+[TestTag("Routing")]
+public class NullableValueTypeOptionTests
+{
+  [ModuleInitializer]
+  internal static void Register() => RegisterTests<NullableValueTypeOptionTests>();
+
+  public static async Task Should_parse_nullable_long_option_with_value()
+  {
+    // Arrange
+    using TestTerminal terminal = new();
+    NuruApp app = NuruApp.CreateBuilder()
+      .UseTerminal(terminal)
+      .Map("test --id {id:long?}")
+      .WithHandler((long? id) => $"id:{id}")
+      .AsCommand()
+      .Done()
+      .Build();
+
+    // Act
+    int exitCode = await app.RunAsync(["test", "--id", "123"]);
+
+    // Assert
+    exitCode.ShouldBe(0);
+    terminal.OutputContains("id:123").ShouldBeTrue();
+
+    await Task.CompletedTask;
+  }
+
+  public static async Task Should_parse_nullable_int_option_with_value()
+  {
+    // Arrange
+    using TestTerminal terminal = new();
+    NuruApp app = NuruApp.CreateBuilder()
+      .UseTerminal(terminal)
+      .Map("test --count {count:int?}")
+      .WithHandler((int? count) => $"count:{count}")
+      .AsCommand()
+      .Done()
+      .Build();
+
+    // Act
+    int exitCode = await app.RunAsync(["test", "--count", "42"]);
+
+    // Assert
+    exitCode.ShouldBe(0);
+    terminal.OutputContains("count:42").ShouldBeTrue();
+
+    await Task.CompletedTask;
+  }
+
+  public static async Task Should_return_null_for_nullable_long_option_without_value()
+  {
+    // Arrange
+    using TestTerminal terminal = new();
+    NuruApp app = NuruApp.CreateBuilder()
+      .UseTerminal(terminal)
+      .Map("test --id? {id:long?}")
+      .WithHandler((long? id) => $"id:{id?.ToString(CultureInfo.InvariantCulture) ?? "null"}")
+      .AsCommand()
+      .Done()
+      .Build();
+
+    // Act
+    int exitCode = await app.RunAsync(["test"]);
+
+    // Assert
+    exitCode.ShouldBe(0);
+    terminal.OutputContains("id:null").ShouldBeTrue();
+
+    await Task.CompletedTask;
+  }
+
+  public static async Task Should_error_on_invalid_nullable_long_option_value()
+  {
+    // Arrange
+    using TestTerminal terminal = new();
+    NuruApp app = NuruApp.CreateBuilder()
+      .UseTerminal(terminal)
+      .Map("test --id {id:long?}")
+      .WithHandler((long? id) => $"id:{id}")
+      .AsCommand()
+      .Done()
+      .Build();
+
+    // Act
+    int exitCode = await app.RunAsync(["test", "--id", "notanumber"]);
+
+    // Assert
+    exitCode.ShouldBe(1);
+    terminal.OutputContains("Error").ShouldBeTrue();
+
+    await Task.CompletedTask;
+  }
+
+  public static async Task Should_parse_nullable_double_option_with_value()
+  {
+    // Arrange
+    using TestTerminal terminal = new();
+    NuruApp app = NuruApp.CreateBuilder()
+      .UseTerminal(terminal)
+      .Map("test --rate {rate:double?}")
+      .WithHandler((double? rate) => $"rate:{rate}")
+      .AsCommand()
+      .Done()
+      .Build();
+
+    // Act
+    int exitCode = await app.RunAsync(["test", "--rate", "3.14"]);
+
+    // Assert
+    exitCode.ShouldBe(0);
+    terminal.OutputContains("rate:3.14").ShouldBeTrue();
+
+    await Task.CompletedTask;
+  }
+
+  public static async Task Should_parse_nullable_bool_option_with_value()
+  {
+    // Arrange
+    using TestTerminal terminal = new();
+    NuruApp app = NuruApp.CreateBuilder()
+      .UseTerminal(terminal)
+      .Map("test --flag {flag:bool?}")
+      .WithHandler((bool? flag) => $"flag:{flag}")
+      .AsCommand()
+      .Done()
+      .Build();
+
+    // Act
+    int exitCode = await app.RunAsync(["test", "--flag", "true"]);
+
+    // Assert
+    exitCode.ShouldBe(0);
+    terminal.OutputContains("flag:True").ShouldBeTrue();
+
+    await Task.CompletedTask;
+  }
+
+  public static async Task Should_return_null_for_nullable_bool_option_without_value()
+  {
+    // Arrange
+    using TestTerminal terminal = new();
+    NuruApp app = NuruApp.CreateBuilder()
+      .UseTerminal(terminal)
+      .Map("test --flag? {flag:bool?}")
+      .WithHandler((bool? flag) => $"flag:{(flag.HasValue ? flag.Value.ToString() : "null")}")
+      .AsCommand()
+      .Done()
+      .Build();
+
+    // Act
+    int exitCode = await app.RunAsync(["test"]);
+
+    // Assert
+    exitCode.ShouldBe(0);
+    terminal.OutputContains("flag:null").ShouldBeTrue();
+
+    await Task.CompletedTask;
+  }
+
+  public static async Task Should_parse_nullable_guid_option_with_value()
+  {
+    // Arrange
+    using TestTerminal terminal = new();
+    NuruApp app = NuruApp.CreateBuilder()
+      .UseTerminal(terminal)
+      .Map("test --guid {guid:guid?}")
+      .WithHandler((Guid? guid) => $"guid:{guid}")
+      .AsCommand()
+      .Done()
+      .Build();
+
+    // Act
+    int exitCode = await app.RunAsync(["test", "--guid", "12345678-1234-1234-1234-123456789012"]);
+
+    // Assert
+    exitCode.ShouldBe(0);
+    terminal.OutputContains("guid:12345678-1234-1234-1234-123456789012").ShouldBeTrue();
+
+    await Task.CompletedTask;
+  }
+
+  public static async Task Should_parse_nullable_datetime_option_with_value()
+  {
+    // Arrange
+    using TestTerminal terminal = new();
+    NuruApp app = NuruApp.CreateBuilder()
+      .UseTerminal(terminal)
+      .Map("test --date {date:datetime?}")
+      .WithHandler((DateTime? date) => $"date:{date:yyyy-MM-dd}")
+      .AsCommand()
+      .Done()
+      .Build();
+
+    // Act
+    int exitCode = await app.RunAsync(["test", "--date", "2024-01-15"]);
+
+    // Assert
+    exitCode.ShouldBe(0);
+    terminal.OutputContains("date:2024-01-15").ShouldBeTrue();
+
+    await Task.CompletedTask;
+  }
+
+  public static async Task Should_handle_multiple_nullable_options()
+  {
+    // Arrange
+    using TestTerminal terminal = new();
+    NuruApp app = NuruApp.CreateBuilder()
+      .UseTerminal(terminal)
+      .Map("test --id? {id:long?} --count? {count:int?}")
+      .WithHandler((long? id, int? count) => $"id:{id?.ToString(CultureInfo.InvariantCulture) ?? "null"}|count:{count?.ToString(CultureInfo.InvariantCulture) ?? "null"}")
+      .AsCommand()
+      .Done()
+      .Build();
+
+    // Act
+    int exitCode = await app.RunAsync(["test", "--id", "123"]);
+
+    // Assert
+    exitCode.ShouldBe(0);
+    terminal.OutputContains("id:123").ShouldBeTrue();
+    terminal.OutputContains("count:null").ShouldBeTrue();
+
+    await Task.CompletedTask;
+  }
+}
+
+} // namespace TimeWarp.Nuru.Tests.Routing

